### PR TITLE
106X backport of: Pile-Up ID training and WPs for 2017 UL samples

### PR DESCRIPTION
**Pull request by marinakolosova**
#### PR description:

This PR is a backport to 106X release. 
The original PR is [#30773](https://github.com/cms-sw/cmssw/pull/30773) and is already merged to the master branch.

There are two commits, the first (2f6a515) includes the actual changes for the UL17 WPs inclusion and the second  
(0d2effe) includes the automatic changes after:
scram build code-checks
scram build code-format
which introduces the large number of lines added/changed with respect to #30773. 
After the second commit, the files:
  RecoJets/JetProducers/interface/PileupJetIdAlgo.h and
  RecoJets/JetProducers/src/PileupJetIdAlgo.cc 
are now identical to the ones in the master branch.

FYI: @alefisico @camclean @singh-ramanpreet

### CODE REVIEW

**Comment by silviodonato** [comment]
please test

**Comment by silviodonato** [comment]
@marinakolosova could you briefly explain in the PR description the differences between this backport (+974 lines added/changed) and #30773 (+171 lines added/changed)? Thanks

**Comment by singh-ramanpreet** [comment]
the tests need these weight files (recently merged): [cms-data/RecoJets-JetProducers/pull/11](https://github.com/cms-data/RecoJets-JetProducers/pull/11)

**Comment by perrotta** [comment]
backport of #30773

**Comment by perrotta** [comment]
please test with  cms-data/RecoJets-JetProducers/pull/11

**Comment by slava77** [comment]
please test with cms-data/RecoJets-JetProducers#11

File: RecoJets/JetProducers/python/PileupJetID_cfi.py [hunk 1]
```diff
@@ -13,8 +13,9 @@
 _chsalgos_81x = cms.VPSet(full_81x_chs,cutbased)
 _chsalgos_94x = cms.VPSet(full_94x_chs,cutbased)
 _chsalgos_102x = cms.VPSet(full_102x_chs,cutbased)
+_chsalgos_106X_UL17 = cms.VPSet(full_106x_UL17_chs,cutbased)
 
-_stdalgos    = _chsalgos_81x
+_stdalgos    = _chsalgos_106X_UL17
```
Comment (slava77) [inline RecoJets/JetProducers/python/PileupJetID_cfi.py:18]: I suppose, this is the origin of the differences in the tests

**Comment by slava77** [comment]
> Reco comparison results: 45 differences found in the comparisons

this PR changes results in many workflows, including the production-like workflows, which is not allowed in a production release.
Please use run2_miniAOD_devel appropriately

**Comment by marinakolosova** [comment]
> > Reco comparison results: 45 differences found in the comparisons
> 
> this PR changes results in many workflows, including the production-like workflows, which is not allowed in a production release.
> Please use run2_miniAOD_devel appropriately


Hi @slava77,
 
 Should the code pick up the 81x default training files? or the 106X?
 I tried the following ways and 1-4 pick the 80X training files and only 5 the 106X. 

1]  80X weight files
_stdalgos    = _chsalgos_81x
run2_miniAOD_devel.toReplaceWith(pileupJetId.algos, _chsalgos_106X_UL17)

 2] 80X weight files
_stdalgos    = _chsalgos_81x
run2_miniAOD_devel.toReplaceWith(pileupJetId.algos, cms.VPSet(_chsalgos_106X_UL17))
 
3] 80X weight files
_stdalgos    = _chsalgos_81x
run2_miniAOD_devel.toModify(pileupJetId, algos = _chsalgos_106X_UL17)
 
4] 80X weight files
_stdalgos    = _chsalgos_81x
run2_miniAOD_devel.toModify(pileupJetId, algos = cms.VPSet(_chsalgos_106X_UL17))
 
5] 106X weight files
_stdalgos    = _chsalgos_106X_UL17
run2_miniAOD_devel.toModify(pileupJetId, algos = _chsalgos_106X_UL17)

 From what I understand, the 3rd is what you are suggesting, right?

@singh-ramanpreet @alefisico @camclean @perrotta

**Comment by slava77** [comment]
> From what I understand, the 3rd is what you are suggesting, right?

right, at least visually (without running the code)

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
please abort test

**Comment by slava77** [comment]
test parameters:
- pull_request = cms-data/RecoJets-JetProducers#11

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by mrodozov** [comment]
@slava77  Do you need a backport of https://github.com/cms-sw/cmsdist/pull/6104 to 10_6 ?

**Comment by slava77** [comment]
> @slava77 Do you need a backport of [cms-sw/cmsdist#6104](https://github.com/cms-sw/cmsdist/pull/6104) to 10_6 ?

yes, please
It looks like the update will be simple,
the difference between the master and 106X in this cms-data repo is just one needed PR
https://github.com/cms-data/RecoJets-JetProducers/compare/V05-11-00...V05-12-00

**Comment by slava77** [comment]
test parameters:
- pull_request = cms-sw/cmsdist#6137

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
> > @slava77 Do you need a backport of [cms-sw/cmsdist#6104](https://github.com/cms-sw/cmsdist/pull/6104) to 10_6 ?
> 
> yes, please
> It looks like the update will be simple,
> the difference between the master and 106X in this cms-data repo is just one needed PR
> [cms-data/RecoJets-JetProducers@V05-11-00...V05-12-00](https://github.com/cms-data/RecoJets-JetProducers/compare/V05-11-00...V05-12-00)

@mrodozov 
I created cms-sw/cmsdist#6137
(it looks like it is not a duplicate already)

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/30935	5b4f49d
- code changes are in line with the PR description and the follow up review
- the backport is acceptable: 
    - differences introduced in #30773 are included, except from the part that changes the defaults, which is done via ```run2_miniAOD_devel.toModify```
    - RecoJets/JetProducers/interface/PileupJetIdAlgo.h and RecoJets/JetProducers/src/PileupJetIdAlgo.cc additionally differ by the results of code-checks
- jenkins tests pass and comparisons with the baseline show no differences (in line with the no-change policy for 106X)

**Comment by silviodonato** [comment]
+1

**Pull request merged**